### ISSUE TITLE
segfaults

### ISSUE THREAD

**Issue by Laeeth** [Dicebot/mood#14]
Thanks, Dicebot, for writing this.  It was on my list of projects to tidy up the vibed.org code (I didn't like depending on a database for this simple task either).

I have run your dicebot.lv branch on two different arch linux x86-64 machines and it segfaults when I try to retrieve a page (just the index works, but only for a bit).  I will look into it further when time, but just to let you know for now.

Laeeth.

**Comment by Dicebot** [comment]
Interesting. Can you possibly provide gdb backtrace for the crash? My actual blog is also running on Arch x64_86 server so I doubt it is a platform issue.

**Comment by Laeeth** [comment]
[laeeth@engine dicebotmood]$ gdb mood
GNU gdb (GDB) 7.9.1
Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-unknown-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from mood...done.
(gdb) run
Starting program: /hgst/dlang/dicebotmood/mood 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
Preparing blog post data
Looking for blog post sources at ./generated/md/
4 posts loaded
Listening for HTTPS requests on ::1:8081
Listening for HTTPS requests on 127.0.0.1:8081
Handling of connection failed: Failed to accept SSL tunnel: error:1407609C:SSL routines:SSL23_GET_CLIENT_HELLO:http request (336027804)
Handling of connection failed: Failed to accept SSL tunnel: error:1407609C:SSL routines:SSL23_GET_CLIENT_HELLO:http request (336027804)
Handling of connection failed: Failed to accept SSL tunnel: error:1407609C:SSL routines:SSL23_GET_CLIENT_HELLO:http request (336027804)
Handling of connection failed: Failed to accept SSL tunnel: error:1407609C:SSL routines:SSL23_GET_CLIENT_HELLO:http request (336027804)
Handling of connection failed: Failed to accept SSL tunnel: error:1407609C:SSL routines:SSL23_GET_CLIENT_HELLO:http request (336027804)
[New Thread 0x7fffc77fe700 (LWP 822)]
[New Thread 0x7fffc7fff700 (LWP 820)]
[New Thread 0x7fffecfba700 (LWP 819)]
[New Thread 0x7fffedffb700 (LWP 818)]
[New Thread 0x7fffee7fc700 (LWP 817)]
[New Thread 0x7fffeeffd700 (LWP 816)]
[New Thread 0x7fffef7fe700 (LWP 815)]
[New Thread 0x7fffeffff700 (LWP 814)]
[New Thread 0x7ffff4fba700 (LWP 813)]
[New Thread 0x7ffff57bb700 (LWP 812)]
[New Thread 0x7fffed7bb700 (LWP 811)]
[New Thread 0x7ffff5fbc700 (LWP 810)]

Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff4fba700 (LWP 813)]
0x0000000000832ff1 in onSocketEvent (buf_event=0x7fffb8003988, status=17, 
    arg=0x7fffb8003010)
    at ../../../home/laeeth/.dub/packages/vibe-d-0.7.24/source/vibe/core/drivers/libevent2_tcp.d:687
687					else ctx.core.resumeTask(ctx.writeOwner, ex);
(gdb) backtrace
#0  0x0000000000832ff1 in onSocketEvent (buf_event=0x7fffb8003988, status=17, 
    arg=0x7fffb8003010)
    at ../../../home/laeeth/.dub/packages/vibe-d-0.7.24/source/vibe/core/drivers/libevent2_tcp.d:687
#1  0x00007ffff79aab66 in ?? () from /usr/lib/libevent-2.0.so.5
#2  0x00007ffff799fca6 in event_base_loop () from /usr/lib/libevent-2.0.so.5
#3  0x0000000000828e97 in vibe.core.drivers.libevent2.Libevent2Driver.runEventLoop() (this=0x7ffff7ee9d00)
    at ../../../home/laeeth/.dub/packages/vibe-d-0.7.24/source/vibe/core/drivers/libevent2.d:181
#4  0x0000000000824823 in vibe.core.core.runEventLoop() ()
    at ../../../home/laeeth/.dub/packages/vibe-d-0.7.24/source/vibe/core/core.d:97
#5  0x00000000008e4c38 in vibe.core.core.workerThreadFunc() ()
    at ../../../home/laeeth/.dub/packages/vibe-d-0.7.24/source/vibe/core/core.d:1487
#6  0x00000000009571dc in core.thread.Thread.run() ()
#7  0x0000000000956ec4 in thread_entryPoint ()
#8  0x00007ffff70884a4 in start_thread () from /usr/lib/libpthread.so.0
#9  0x00007ffff68c012d in clone () from /usr/lib/libc.so.6

**Comment by Laeeth** [comment]
(Beginning was just because I forgot to use https)

**Comment by Laeeth** [comment]
fwiw it _crashes_ ldc (I have no idea how to interpret LLVM errors) but compiles and works fine under gdc.  So I can use it anyway.  I had to put some conditional code based on version in for std.string:format vs std.format:format, and paste in lineSplitter and chunkyBy from Phobos, but it works.  I imagine you don't need trouble for small project but can give pull request if helpful.

This is the ldc compiler crash:
Target vibe-d 0.7.24 is up to date. Use --force to rebuild.
Building mood 1.0.0+commit.4.g4a5a799 configuration "application", build type debug.
Running ldc...
2066L
Compiling diet template 'pages/new_post.dt'...
Compiling diet template 'pages/index.dt'...
Compiling diet template 'pages/single_post.dt'...
0  ldc             0x00000000012335f2 llvm::sys::PrintStackTrace(_IO_FILE*) + 34
1  ldc             0x00000000012316a3
2  libpthread.so.0 0x00007f9fb96f6d60
3  ldc             0x00000000006ea97f
4  ldc             0x00000000006eb6e3 IrAggr::addFieldInitializers(llvm::SmallVectorImpl<llvm::Constant*>&, std::map<VarDeclaration*, llvm::Constant*, std::less<VarDeclaration*>, std::allocator<std::pair<VarDeclaration* const, llvm::Constant*> > > const&, AggregateDeclaration*, unsigned int&, bool) + 707
5  ldc             0x00000000006ebc74 IrAggr::createInitializerConstant(std::map<VarDeclaration*, llvm::Constant*, std::less<VarDeclaration*>, std::allocator<std::pair<VarDeclaration* const, llvm::Constant*> > > const&, llvm::StructType*) + 164
6  ldc             0x00000000006ec0c7 IrAggr::getDefaultInit() + 119
7  ldc             0x0000000000732ba8
8  ldc             0x0000000000732f33
9  ldc             0x0000000000732b83
10 ldc             0x000000000073373e Declaration_codegen(Dsymbol*) + 46
11 ldc             0x00000000006c5b19 codegenModule(IRState*, Module*, bool) + 137
12 ldc             0x000000000059b631 ldc::CodeGenerator::emit(Module*) + 177
13 ldc             0x000000000057690b main + 10395
14 libc.so.6       0x00007f9fb88c8610 __libc_start_main + 240
15 ldc             0x0000000000594365
FAIL .dub/build/application-debug-linux.posix-x86_64-ldc_0-8E7665FF038D65A1E991C5F79E7FBA61/ mood executable
Error executing command build:
ldc failed with exit code -11.

**Comment by Dicebot** [comment]
Aha, it must have worked for me fine because my currently deployed version is build with dmd.
Anyway, the crash trace doesn't seem to even mention single line of actual mood sources so I am inclined to think it is the problem with vibe.d / ldc.

FYI @s-ludwig @klickverbot

**Comment by Laeeth** [comment]
Sorry if I wasnt clear.  It always segfaults on dmd, crashes the ldc compiler, but kind of works on gdc.

I rewrote it and it works for me now, but I am not quite sure what the problem was.   It seemed to crash in the render delegate.  I tore out the generic cache and simplified it, and changed variable naming convention to Phobos so not sure if a pull request will be welcome;). But I am very grateful to you for writing, as it saves me time.

I wrote another little script that may be helpful that takes Nikola style .md and .meta files and combines them into a single .md file.  That makes it easy to import from Wordpress as Nikola does this okay.
